src/components/PaginationV2/PaginationV2.js

```diff
@@ -0,0 +1,228 @@
+    disabled: false,
+    page: 1,
+    pagesUnknown: false,
+    isLastPage: false,
+    pageInputDisabled: false,
+    itemText: (min, max) => `${min}-${max} items`,
+    pageText: page => `page ${page}`,
+  };
+
+  state = {
+    page: this.props.page,
+    pageSize:
+      this.props.pageSize && this.props.pageSizes.includes(this.props.pageSize)
+        ? this.props.pageSize
+        : this.props.pageSizes[0],
+  };
+
+  componentWillMount() {
+    this.uniqueId = `${Math.floor(Math.random() * 0xffff)}`;
+  }
+
+  componentWillReceiveProps({ pageSizes, page, pageSize }) {
+    if (!equals(pageSizes, this.props.pageSizes)) {
+      this.setState({ pageSize: pageSizes[0], page: 1 });
+    }
+    if (page !== this.props.page) {
+      this.setState({
+        page,
+      });
+    }
+    if (pageSize !== this.props.pageSize) {
+      this.setState({ pageSize });
+    }
+  }
+
+  handleSizeChange = evt => {
+    const pageSize = Number(evt.target.value);
+    this.setState({ pageSize, page: 1 });
+    this.props.onChange({ page: 1, pageSize });
+  };
+
+  handlePageChange = evt => {
+    this.setState({ page: evt.target.value });
+  };
+
+  handlePageInputChange = evt => {
+    const page = Number(evt.target.value);
+    if (
+      page > 0 &&
+      page <= Math.ceil(this.props.totalItems / this.state.pageSize)
+    ) {
+      this.setState({ page });
+      this.props.onChange({ page, pageSize: this.state.pageSize });
+    }
+  };
+
+  incrementPage = () => {
+    const page = this.state.page + 1;
+    this.setState({ page });
+    this.props.onChange({ page, pageSize: this.state.pageSize });
+  };
+
+  decrementPage = () => {
+    const page = this.state.page - 1;
+    this.setState({ page });
+    this.props.onChange({ page, pageSize: this.state.pageSize });
+  };
+
+  renderSelectItems = total => {
+    let counter = 1;
+    let itemArr = [];
+    while (counter <= total) {
+      itemArr.push(
+        <SelectItem key={counter} value={counter} text={String(counter)} />
+      );
+      counter++;
+    }
+    return itemArr;
+  };
+
+  render() {
+    const {
+      backwardText,
+      className,
+      forwardText,
+      id,
+      itemsPerPageText,
+      itemRangeText,
+      pageRangeText,
+      pageSize, // eslint-disable-line no-unused-vars
+      pageSizes,
+      itemText,
+      pageText,
+      pageNumberText, // eslint-disable-line no-unused-vars
+      pagesUnknown,
+      isLastPage,
+      pageInputDisabled,
+      totalItems,
+      onChange, // eslint-disable-line no-unused-vars
+      page: pageNumber, // eslint-disable-line no-unused-vars
+      ...other
+    } = this.props;
+
+    const statePage = this.state.page;
+    const statePageSize = this.state.pageSize;
+    const classNames = classnames('bx--pagination', className);
+    const inputId = id || this.uniqueId;
+    const totalPages = Math.ceil(totalItems / statePageSize);
+    const selectItems = this.renderSelectItems(totalPages);
+
+    return (
```

Comment:
Could we break up `Pagination` into smaller components, or does it make more sense for it to be in one file?

Reply:
That could probably be a good refactor project for v9, or an enhancement for when the tables are ready.